src/test/java/LocationsUtilityTest.java

```diff
@@ -0,0 +1,121 @@
+
+  private static final List<Task> NO_TASKS = ImmutableList.of();
+  private static final List<Task> TASKS_WITH_NO_NOTES = ImmutableList.of(TASK_WITH_NO_NOTES);
+  private static final List<Task> TASKS_WITH_NO_LOCATION = ImmutableList.of(TASK_WITH_NO_LOCATION);
+  private static final List<Task> TASKS_WITH_ONE_LOCATION =
+      ImmutableList.of(TASK_WITH_ONE_LOCATION);
+  private static final List<Task> TASKS_WITH_TWO_LOCATIONS =
+      ImmutableList.of(TASK_WITH_TWO_LOCATIONS);
+  private static final List<Task> TASKS_WITH_ONE_LOCATION_ENCLOSED_INCORRECTLY =
+      ImmutableList.of(TASK_WITH_ONE_LOCATION_ENCLOSED_INCORRECTLY);
+  private static final List<Task> TASKS_WITH_EMPTY_LOCATION =
+      ImmutableList.of(TASK_WITH_EMPTY_LOCATION);
+  private static final List<Task> ALL_TASKS =
+      ImmutableList.of(
+          TASK_WITH_NO_NOTES,
+          TASK_WITH_NO_LOCATION,
+          TASK_WITH_ONE_LOCATION,
+          TASK_WITH_TWO_LOCATIONS,
+          TASK_WITH_ONE_LOCATION_ENCLOSED_INCORRECTLY,
+          TASK_WITH_EMPTY_LOCATION);
+
+  @Test
+  public void getLocationNoTasks() {
+    // Obtain locations in the the task notes of no tasks.
+    Assert.assertEquals(ImmutableList.of(), LocationsUtility.getLocations(PREFIX, NO_TASKS));
+  }
+
+  @Test
+  public void getLocationNoNotes() {
+    // Obtain location in the task notes of one task with no notes defined.
+    Assert.assertEquals(
+        ImmutableList.of(), LocationsUtility.getLocations(PREFIX, TASKS_WITH_NO_NOTES));
+  }
+
+  @Test
+  public void getNoLocation() {
+    // Obtain location in the task notes of one task with no location.
+    Assert.assertEquals(
+        ImmutableList.of(), LocationsUtility.getLocations(PREFIX, TASKS_WITH_NO_LOCATION));
+  }
+
+  @Test
+  public void getOneLocation() {
+    // Obtain location in the task notes of one task with one location.
+    Assert.assertEquals(
+        ImmutableList.of("Google Kitchener"),
+        LocationsUtility.getLocations(PREFIX, TASKS_WITH_ONE_LOCATION));
+  }
+
+  @Test
+  public void getTwoLocations() {
+    // Obtain location in the task notes of one task with two locations. Second location, Google
+    // Montreal, is ignored.
+    Assert.assertEquals(
+        ImmutableList.of("Google Kitchener"),
+        LocationsUtility.getLocations(PREFIX, TASKS_WITH_TWO_LOCATIONS));
+  }
+
+  @Test
+  public void getIncorrectlyEnclosedLocation() {
+    // Obtain location in the task notes of one task with one location enclosed with () instead of
+    // [].
+    Assert.assertEquals(
+        ImmutableList.of(),
+        LocationsUtility.getLocations(PREFIX, TASKS_WITH_ONE_LOCATION_ENCLOSED_INCORRECTLY));
+  }
+
+  @Test
+  public void getEmptyLocation() {
+    // Obtain location in the task notes of one task with one location with [Location: ] tag but
+    // nothing inside of it.
+    Assert.assertEquals(
+        ImmutableList.of(""), LocationsUtility.getLocations(PREFIX, TASKS_WITH_EMPTY_LOCATION));
```

Comment:
Is empty location really a valid location? Or is it a special case of a missing/malformed location?

(Maybe there is a valid use case for empty locations - it depends on how this utility will be used.)

Reply:
No it is not a valid location, the utility's job is to just parse and the validation should be done in the servlet but since the "No prefix" is changed to "", these are now filtered as well. Thanks for pointing it out!